### ISSUE TITLE
Generic incorrectly caught as `list[unknown]` rather than `list[T]` in python 3.12 in dataclass delaration when using list factory

### ISSUE THREAD

**Issue by Don-Burns** [microsoft/pyright#10483]
## Environment data

-   Pylance version: 2025.5.1
-   OS and version: Windows 11 (using WSL2 Ubuntu 22.04.5 LTS)
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.12.10 from deadsnakes ppa

## Code Snippet

```python
from dataclasses import dataclass, field


@dataclass
class Dataclass[T]:
    entries: list[T] = field(default_factory=list) # Type of "entries" is partially unknown
                                                                      #      Type of "entries" is "list[Unknown]"Pylance[reportUnknownVariableType]
                                                                      # (variable) entries: list[T@Dataclass]


val = Dataclass(entries=[1, 2, 3])

reveal_type(val.entries)  # Type of "val.entries" is "list[int]" Pylance
```

![Image](https://github.com/user-attachments/assets/133db788-fa59-4ef5-8735-023967b6837c)

## Repro Steps

1. open script using the versions of the extension and python provided above

## Expected behavior

From what I know this is a valid declaration of a Generic with 3.12. Mypy doesn't complain under strict flags and pylance doesn't complain with usage and gives correct type after object instantiation

## Actual behavior

Incorrectly complaining about unknown type of `entries` generic list when using dataclass.field

## Logs

```
2025-05-21 15:30:07.700 [info] (Client) Pylance async client (2025.5.1) started with python extension (2025.6.1)
2025-05-21 15:30:08.667 [info] [Info  - 3:30:08 PM] (12275) Server root directory: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist
2025-05-21 15:30:08.669 [info] [Info  - 3:30:08 PM] (12275) Pylance language server 2025.5.1 (pyright version 1.1.400, commit b962ffc2) starting
2025-05-21 15:30:08.680 [info] [Info  - 3:30:08 PM] (12275) Starting service instance "test" for workspace "/home/dburns/src/test"
2025-05-21 15:30:08.765 [info] [Info  - 3:30:08 PM] (12275) Setting environmentName for service "test": "3.12.10 (.venv venv)"
2025-05-21 15:30:08.766 [info] [Info  - 3:30:08 PM] (12275) Setting pythonPath for service "test": "/home/dburns/src/test/.venv/bin/python"
2025-05-21 15:30:08.766 [info] [Info  - 3:30:08 PM] (12275) No include entries specified; assuming /home/dburns/src/test
2025-05-21 15:30:08.767 [info] [Info  - 3:30:08 PM] (12275) Auto-excluding **/node_modules
2025-05-21 15:30:08.767 [info] [Info  - 3:30:08 PM] (12275) Auto-excluding **/__pycache__
2025-05-21 15:30:08.768 [info] [Info  - 3:30:08 PM] (12275) Auto-excluding **/.*
2025-05-21 15:30:08.786 [info] [Info  - 3:30:08 PM] (12275) Assuming Python version 3.12.10.final.0
2025-05-21 15:30:08.934 [info] [Info  - 3:30:08 PM] (12275) Found 1 source file
2025-05-21 15:30:08.973 [info] [Info  - 3:30:08 PM] (12275) Setting environmentName for service "test": "3.12.10 (.venv venv)"
2025-05-21 15:30:08.973 [info] [Info  - 3:30:08 PM] (12275) Setting pythonPath for service "test": "/home/dburns/src/test/.venv/bin/python"
2025-05-21 15:30:08.973 [info] [Info  - 3:30:08 PM] (12275) No include entries specified; assuming /home/dburns/src/test
2025-05-21 15:30:08.973 [info] [Info  - 3:30:08 PM] (12275) Auto-excluding **/node_modules
2025-05-21 15:30:08.973 [info] [Info  - 3:30:08 PM] (12275) Auto-excluding **/__pycache__
2025-05-21 15:30:08.973 [info] [Info  - 3:30:08 PM] (12275) Auto-excluding **/.*
2025-05-21 15:30:08.992 [info] [Info  - 3:30:08 PM] (12275) Assuming Python version 3.12.10.final.0
2025-05-21 15:30:09.027 [info] [Info  - 3:30:09 PM] (12275) Found 1 source file
2025-05-21 15:30:09.300 [info] [Info  - 3:30:09 PM] (12275) BG: Priority queue background worker(2) root directory: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist
2025-05-21 15:30:09.300 [info] [Info  - 3:30:09 PM] (12275) BG: Priority queue background worker(2) started
2025-05-21 15:30:50.476 [info] [Info  - 3:30:50 PM] (12275) Setting environmentName for service "test": "3.12.10 (.venv venv)"
2025-05-21 15:30:50.476 [info] [Info  - 3:30:50 PM] (12275) Setting pythonPath for service "test": "/home/dburns/src/test/.venv/bin/python"
2025-05-21 15:30:50.476 [info] [Info  - 3:30:50 PM] (12275) No include entries specified; assuming /home/dburns/src/test
2025-05-21 15:30:50.476 [info] [Info  - 3:30:50 PM] (12275) Auto-excluding **/node_modules
2025-05-21 15:30:50.476 [info] [Info  - 3:30:50 PM] (12275) Auto-excluding **/__pycache__
2025-05-21 15:30:50.476 [info] [Info  - 3:30:50 PM] (12275) Auto-excluding **/.*
2025-05-21 15:30:50.492 [info] [Info  - 3:30:50 PM] (12275) Assuming Python version 3.12.10.final.0
2025-05-21 15:30:50.529 [info] [Info  - 3:30:50 PM] (12275) Found 1 source file
2025-05-21 15:30:58.500 [info] [Info  - 3:30:58 PM] (12275) Setting environmentName for service "test": "3.12.10 (.venv venv)"
2025-05-21 15:30:58.500 [info] [Info  - 3:30:58 PM] (12275) Setting pythonPath for service "test": "/home/dburns/src/test/.venv/bin/python"
2025-05-21 15:30:58.500 [info] [Info  - 3:30:58 PM] (12275) No include entries specified; assuming /home/dburns/src/test
2025-05-21 15:30:58.500 [info] [Info  - 3:30:58 PM] (12275) Auto-excluding **/node_modules
2025-05-21 15:30:58.500 [info] [Info  - 3:30:58 PM] (12275) Auto-excluding **/__pycache__
2025-05-21 15:30:58.500 [info] [Info  - 3:30:58 PM] (12275) Auto-excluding **/.*
2025-05-21 15:30:58.514 [info] [Info  - 3:30:58 PM] (12275) Assuming Python version 3.12.10.final.0
2025-05-21 15:30:58.637 [info] [Info  - 3:30:58 PM] (12275) Found 1 source file
2025-05-21 15:31:07.339 [info] (12275) No configuration file found.
2025-05-21 15:31:07.339 [info] (12275) No pyproject.toml file found.
2025-05-21 15:31:07.339 [info] [Info  - 3:31:07 PM] (12275) Setting environmentName for service "test": "3.12.10 (.venv venv)"
2025-05-21 15:31:07.339 [info] [Info  - 3:31:07 PM] (12275) Setting pythonPath for service "test": "/home/dburns/src/test/.venv/bin/python"
2025-05-21 15:31:07.339 [info] [Info  - 3:31:07 PM] (12275) No include entries specified; assuming /home/dburns/src/test
2025-05-21 15:31:07.339 [info] [Info  - 3:31:07 PM] (12275) Auto-excluding **/node_modules
2025-05-21 15:31:07.340 [info] [Info  - 3:31:07 PM] (12275) Auto-excluding **/__pycache__
2025-05-21 15:31:07.340 [info] [Info  - 3:31:07 PM] (12275) Auto-excluding **/.*
2025-05-21 15:31:07.357 [info] [Info  - 3:31:07 PM] (12275) Assuming Python version 3.12.10.final.0
2025-05-21 15:31:07.357 [info] (12275) Assuming Python platform Linux
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275) Execution environment: 3.12.10 (.venv venv)
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)   Extra paths:
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)     (none)
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)   Python version: 3.12.10.final.0
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)   Python platform: Linux
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)   Search paths:
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)     /home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)     /home/dburns/src/test
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)     /home/dburns/src/test/typings
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)     /home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stubs/...
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)     /home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/bundled/stubs
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)     /usr/lib/python3.12
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)     /usr/lib/python3.12/lib-dynload
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275)     /home/dburns/src/test/.venv/lib/python3.12/site-packages
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275) Adding fs watcher for library directories:
 file:///usr/lib/python3.12
file:///home/dburns/src/test/.venv/lib/python3.12/site-packages
2025-05-21 15:31:07.393 [info] [Info  - 3:31:07 PM] (12275) Adding fs watcher for directories:
 file:///home/dburns/src/test
2025-05-21 15:31:07.394 [info] (12275) Searching for source files
2025-05-21 15:31:07.395 [info] [Info  - 3:31:07 PM] (12275) Found 1 source file
2025-05-21 15:31:07.395 [info] (12275) Adding fs watcher for files:
 file:///home/dburns/src/test/requirements.txt
file:///home/dburns/src/test/pyproject.toml
2025-05-21 15:31:07.399 [info] [Info  - 3:31:07 PM] (12275) SourceFile: Received fs event 'change' for path '/home/dburns/src/test/.vscode/settings.json'
2025-05-21 15:31:07.399 [info] [Info  - 3:31:07 PM] (12275) SourceFile: Received fs event 'change' for path '/home/dburns/src/test/.vscode/settings.json'
2025-05-21 15:31:07.399 [info] [Info  - 3:31:07 PM] (12275) SourceFile: Received fs event 'change' for path '/home/dburns/src/test/.vscode/settings.json'
2025-05-21 15:31:07.407 [info] (12275) pytest configurations: {"message":"script","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}
2025-05-21 15:31:07.413 [info] (12275) BG: pytest configurations: {"message":"script","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}
2025-05-21 15:31:11.828 [info] [Info  - 3:31:11 PM] (12275) SourceFile: Received fs event 'change' for path '/home/dburns/src/test/.vscode/launch.json'
2025-05-21 15:31:11.828 [info] [Info  - 3:31:11 PM] (12275) SourceFile: Received fs event 'change' for path '/home/dburns/src/test/.vscode/launch.json'
2025-05-21 15:31:11.828 [info] [Info  - 3:31:11 PM] (12275) SourceFile: Received fs event 'change' for path '/home/dburns/src/test/.vscode/launch.json'
2025-05-21 15:31:14.439 [info] [Info  - 3:31:14 PM] (12275) SourceFile: Received fs event 'change' for path '/home/dburns/src/test/.vscode/tasks.json'
2025-05-21 15:31:14.439 [info] [Info  - 3:31:14 PM] (12275) SourceFile: Received fs event 'change' for path '/home/dburns/src/test/.vscode/tasks.json'
2025-05-21 15:31:14.439 [info] [Info  - 3:31:14 PM] (12275) SourceFile: Received fs event 'change' for path '/home/dburns/src/test/.vscode/tasks.json'
2025-05-21 15:31:15.649 [info] (12275) Attempting to resolve using local imports: __builtins__
2025-05-21 15:31:15.692 [info] (12275) [FG] parsing: file:///home/dburns/src/test/test.py (36ms)
2025-05-21 15:31:15.708 [info] (12275) [FG] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 3ms] (87ms)
2025-05-21 15:31:15.717 [info] (12275) [FG] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi (35ms)
2025-05-21 15:31:15.748 [info] (12275) [FG] binding: file:///home/dburns/src/test/test.py (1ms)
2025-05-21 15:31:15.815 [info] (12275) BG: Attempting to resolve using local imports: __builtins__
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] analyzing: file:///home/dburns/src/test/test.py ...
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)]   parsing: file:///home/dburns/src/test/test.py (47ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)]   parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 1ms] (76ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)]   binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi (34ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)]   binding: file:///home/dburns/src/test/test.py (0ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)]   checking: file:///home/dburns/src/test/test.py (34ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] analyzing: file:///home/dburns/src/test/test.py (194ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] SemanticTokens full at file:///home/dburns/src/test/test.py (0ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/typing.pyi [fs read 1ms] (16ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/typing.pyi (14ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/types.pyi [fs read 3ms] (10ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/types.pyi (11ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 1ms] (11ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (4ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/abc.pyi [fs read 0ms] (1ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/abc.pyi (2ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/dataclasses.pyi [fs read 1ms] (4ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/dataclasses.pyi (1ms)
2025-05-21 15:31:15.815 [info] (12275) BG: [BG(1)] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/collections/abc.pyi [fs read 0ms] (1ms)
2025-05-21 15:31:15.816 [info] (12275) BG: [BG(1)] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/collections/abc.pyi ...
2025-05-21 15:31:15.816 [info] (12275) BG: [BG(1)]   parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/_collections_abc.pyi [fs read 0ms] (3ms)
2025-05-21 15:31:15.816 [info] (12275) BG: [BG(1)]   binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/_collections_abc.pyi (2ms)
2025-05-21 15:31:15.816 [info] (12275) BG: [BG(1)] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/collections/abc.pyi (6ms)
2025-05-21 15:31:15.816 [info] (12275) BG: [BG(1)] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/enum.pyi [fs read 1ms] (5ms)
2025-05-21 15:31:15.816 [info] (12275) BG: [BG(1)] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/enum.pyi (3ms)
2025-05-21 15:31:15.816 [info] (12275) BG: [BG(1)] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 2ms] (10ms)
2025-05-21 15:31:15.816 [info] (12275) BG: [BG(1)] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/typing_extensions.pyi (3ms)
2025-05-21 15:31:15.816 [info] [Info  - 3:31:15 PM] (12275) BG: Indexer background runner(3) root directory: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist (index)
2025-05-21 15:31:15.816 [info] [Info  - 3:31:15 PM] (12275) BG: Indexing(3) started
2025-05-21 15:31:15.819 [info] (12275) BG: [BG(1)] indexing: file:///home/dburns/src/test/test.py [found 2] (1ms)
2025-05-21 15:31:15.819 [info] (12275) BG: Workspace indexing done: file:///home/dburns/src/test/test.py
2025-05-21 15:31:15.819 [info] [Info  - 3:31:15 PM] (12275) BG: Heap stats: total_memory_size=15720MB, total_free_size=10244MB, total_heap_size=109MB, used_heap_size=82MB, cross_worker_used_heap_size=82MB, total_physical_size=101MB, total_available_size=2012MB, heap_size_limit=2096MB
2025-05-21 15:31:15.820 [info] (12275) BG: [BG(1)] analyzing: file:///home/dburns/src/test/test.py ...
2025-05-21 15:31:15.820 [info] (12275) BG: [BG(1)]   checking: file:///home/dburns/src/test/test.py ...
2025-05-21 15:31:15.821 [info] (12275) BG: [BG(1)]     parsing: file:///usr/lib/python3.12/dataclasses.py [fs read 3ms] (35ms)
2025-05-21 15:31:15.822 [info] (12275) BG: [BG(1)]     binding: file:///usr/lib/python3.12/dataclasses.py (12ms)
2025-05-21 15:31:15.822 [info] (12275) BG: [IDX(3)] scan packages file:///home/dburns/src/test ...
2025-05-21 15:31:15.822 [info] (12275) BG: [IDX(3)]   read stdlib indices (27ms)
2025-05-21 15:31:15.822 [info] (12275) BG: [BG(1)]   checking: file:///home/dburns/src/test/test.py (58ms)
2025-05-21 15:31:15.822 [info] (12275) BG: [BG(1)] analyzing: file:///home/dburns/src/test/test.py (59ms)
2025-05-21 15:31:15.826 [info] (12275) BG: [BG(1)] SemanticTokens range 0:0 - 12:0 at file:///home/dburns/src/test/test.py (0ms)
2025-05-21 15:31:16.018 [info] (12275) BG: [IDX(3)] scan packages file:///home/dburns/src/test (460ms)
2025-05-21 15:31:16.018 [info] [Info  - 3:31:16 PM] (12275) BG: scanned(3) 10 files over 1 exec env
2025-05-21 15:31:16.146 [info] (12275) BG: [IDX(3)] index packages file:///home/dburns/src/test ...
2025-05-21 15:31:16.146 [info] (12275) BG: [IDX(3)]   index execution environment file:///home/dburns/src/test ...
2025-05-21 15:31:16.146 [info] (12275) BG: [IDX(3)]     indexing: file:///usr/lib/python3.12/__phello__/__init__.py [skipped: no '__all__' defined] (1ms)
2025-05-21 15:31:16.170 [info] (12275) [FG] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/typing.pyi [fs read 2ms] (19ms)
2025-05-21 15:31:16.188 [info] (12275) [FG] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/typing.pyi (17ms)
2025-05-21 15:31:16.212 [info] (12275) [FG] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/types.pyi [fs read 0ms] (10ms)
2025-05-21 15:31:16.214 [info] (12275) [FG] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/types.pyi (3ms)
2025-05-21 15:31:16.226 [info] (12275) [FG] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 0ms] (4ms)
2025-05-21 15:31:16.228 [info] (12275) [FG] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (2ms)
2025-05-21 15:31:16.235 [info] (12275) [FG] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/abc.pyi [fs read 0ms] (0ms)
2025-05-21 15:31:16.236 [info] (12275) [FG] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/abc.pyi (0ms)
2025-05-21 15:31:16.244 [info] (12275) [FG] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/dataclasses.pyi [fs read 0ms] (3ms)
2025-05-21 15:31:16.245 [info] (12275) [FG] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/dataclasses.pyi (1ms)
2025-05-21 15:31:16.248 [info] (12275) [FG] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/collections/abc.pyi [fs read 0ms] (1ms)
2025-05-21 15:31:16.249 [info] (12275) [FG] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/collections/abc.pyi ...
2025-05-21 15:31:16.249 [info] (12275) [FG]   parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/_collections_abc.pyi [fs read 0ms] (1ms)
2025-05-21 15:31:16.250 [info] (12275) [FG]   binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/_collections_abc.pyi (1ms)
2025-05-21 15:31:16.250 [info] (12275) [FG] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/collections/abc.pyi (2ms)
2025-05-21 15:31:16.259 [info] (12275) [FG] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/enum.pyi [fs read 0ms] (4ms)
2025-05-21 15:31:16.264 [info] (12275) [FG] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/enum.pyi (2ms)
2025-05-21 15:31:16.277 [info] (12275) [FG] parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 0ms] (10ms)
2025-05-21 15:31:16.278 [info] (12275) [FG] binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/typing_extensions.pyi (2ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]     indexing: file:///usr/lib/python3.12/concurrent/__init__.py [skipped: no '__all__' defined] (1ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]     indexing: file:///usr/lib/python3.12/pydoc_data/__init__.py [skipped: no '__all__' defined] (0ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]     indexing: file:///usr/lib/python3.12/test/__init__.py [skipped: no '__all__' defined] (1ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]     indexing: file:///usr/lib/python3.12/urllib/__init__.py [skipped: no '__all__' defined] (0ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]     indexing: file:///usr/lib/python3.12/wsgiref/__init__.py [skipped: no '__all__' defined] (1ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]     indexing: file:///usr/lib/python3.12/xmlrpc/__init__.py [skipped: no '__all__' defined] (0ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]     indexing: file:///home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__init__.py ...
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]       parsing: file:///home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__init__.py [fs read 0ms] (10ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]       parsing: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 2ms] (52ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]       binding: file:///home/dburns/.vscode-server/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi (12ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]       binding: file:///home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__init__.py (0ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]     indexing: file:///home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__init__.py [found 0] (78ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]     indexing: file:///home/dburns/src/test/.venv/lib/python3.12/site-packages/mypyc/__init__.py ...
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]       parsing: file:///home/dburns/src/test/.venv/lib/python3.12/site-packages/mypyc/__init__.py [fs read 0ms] (1ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]       binding: file:///home/dburns/src/test/.venv/lib/python3.12/site-packages/mypyc/__init__.py (0ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]     indexing: file:///home/dburns/src/test/.venv/lib/python3.12/site-packages/mypyc/__init__.py [found 0] (2ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)]   index execution environment file:///home/dburns/src/test [found 729 in 10 files] (90ms)
2025-05-21 15:31:16.306 [info] (12275) BG: [IDX(3)] index packages file:///home/dburns/src/test [found 729 in 1 exec envs] (100ms)
2025-05-21 15:31:16.306 [info] [Info  - 3:31:16 PM] (12275) BG: indexed(3) 10 files over 1 exec env
2025-05-21 15:31:16.313 [info] [Info  - 3:31:16 PM] (12275) BG: Indexing finished(3).
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) LibraryFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/__pycache__/typing_extensions.cpython-312.pyc'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) SourceFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/__pycache__/typing_extensions.cpython-312.pyc'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) LibraryFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__pycache__'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) SourceFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__pycache__'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) LibraryFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/dmypy/__pycache__'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) SourceFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/dmypy/__pycache__'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) LibraryFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/__pycache__/mypy_extensions.cpython-312.pyc'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) SourceFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/__pycache__/mypy_extensions.cpython-312.pyc'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) LibraryFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__pycache__/bogus_type.cpython-312.pyc'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) SourceFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__pycache__/bogus_type.cpython-312.pyc'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) LibraryFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__pycache__/pyinfo.cpython-312.pyc'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) SourceFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__pycache__/pyinfo.cpython-312.pyc'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) LibraryFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__pycache__/split_namespace.cpython-312.pyc'
2025-05-21 15:31:16.830 [info] [Info  - 3:31:16 PM] (12275) SourceFile: Received fs event 'add' for path '/home/dburns/src/test/.venv/lib/python3.12/site-packages/mypy/__pycache__/split_namespace.cpython-312.pyc'
2025-05-21 15:31:32.451 [info] [Info  - 3:31:32 PM] (12275) SourceFile: Received fs event 'change' for path '/home/dburns/src/test/test.py'
2025-05-21 15:31:32.460 [info] (12275) BG: [BG(1)] indexing: file:///home/dburns/src/test/test.py [found 2] (0ms)
2025-05-21 15:31:32.460 [info] (12275) BG: Workspace indexing done: file:///home/dburns/src/test/test.py
2025-05-21 15:31:32.460 [info] [Info  - 3:31:32 PM] (12275) BG: Heap stats: total_memory_size=15720MB, total_free_size=10236MB, total_heap_size=74MB, used_heap_size=72MB, cross_worker_used_heap_size=72MB, total_physical_size=73MB, total_available_size=2024MB, heap_size_limit=2096MB

```

**Comment by erictraut** [comment]
This is a duplicate of https://github.com/microsoft/pyright/issues/10277.

You can fix this by changing your code to:

```python
    entries: list[T] = field(default_factory=list[T])
```
